### PR TITLE
Feature/heatmap improvements

### PR DESCRIPTION
**Pull request by pacs01**

### CODE REVIEW

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import { inputHandler } from "../../../utils/InputHandler";
+import type { AllInputTypes } from "../../../utils/InputHandler";
+import HeatmapData from "../../../models/HeatmapData";
+import type { HeatmapMode } from "../../../types/Heatmap";
+
+type Props = {
+	heatmapData: HeatmapData,
+	maxLuxValue: number,
+	heatmapMode: HeatmapMode,
+	onChange: (string, AllInputTypes) => void
```
Comment (eluchsinger) [inline src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx]: wrong signature: `onChange: (key: string, value: AllInputTypes) => void`

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	render() {
+		return (
+			<Form heading="Auswertung">
+				<FormFields>
+					<fieldset>
+						<FormField
+							label="Maximale Beleuchtungsstärke einschränken (in Lux)"
+							help={"(0 = kein Limit)"}
```
Comment (eluchsinger) [inline src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx]: Use string directly

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+						>
+							<NumberInput
+								disabled={this.props.heatmapMode !== "DEFAULT"}
+								name={"maxLuxValue"}
```
Comment (eluchsinger) [inline src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx]: Use string directly

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+								value={this.props.maxLuxValue}
+								min={0}
+								step={1}
+								onChange={inputHandler(this.props.onChange)}
+							/>
+						</FormField>
+						<Box direction={"row"} pad={{ between: "medium" }}>
+							<Value
+								label={"Anzahl Messungen"}
```
Comment (eluchsinger) [inline src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx]: Use string directly

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+							<NumberInput
+								disabled={this.props.heatmapMode !== "DEFAULT"}
+								name={"maxLuxValue"}
+								value={this.props.maxLuxValue}
+								min={0}
+								step={1}
+								onChange={inputHandler(this.props.onChange)}
+							/>
+						</FormField>
+						<Box direction={"row"} pad={{ between: "medium" }}>
```
Comment (eluchsinger) [inline src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx]: Use string directly

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+							<Value
+								label={"Anzahl Messungen"}
+								value={this.state.numberOfReadings}
+								size={"small"}
```
Comment (eluchsinger) [inline src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx]: Use string directly

File: src/components/measurements/MeasurementSummary.jsx [hunk 7]
```diff
@@ -110,7 +110,7 @@ export default class MeasurementSummary extends React.Component<Props, State> {
 								configObject={this.state.configObject}
 								heatmapMode={this.state.heatmapMode}
 							/>
-							<Box>
+							<Box basis={"medium"} flex pad={{ horizontal: "medium" }}>
```
Comment (eluchsinger) [inline src/components/measurements/MeasurementSummary.jsx]: Use string directly

File: src/containers/fluxHeatmap/FluxHeatmap.jsx [hunk 8]
```diff
@@ -14,6 +14,8 @@ import { PLACEHOLDER_IMAGE } from "../../images/ImagesBase64";
 import HeatmapLegend from "./HeatmapLegend";
 import HeatmapTooltip from "./HeatmapTooltip";
 import BrowserPosition from "../../models/BrowserPosition";
+import type { AllInputTypes } from "../../utils/InputHandler";
```
Comment (eluchsinger) [inline src/containers/fluxHeatmap/FluxHeatmap.jsx]: Wrong order (type after imports)

File: src/containers/fluxHeatmap/FluxHeatmap.jsx [hunk 9]
```diff
@@ -180,6 +187,8 @@ export default class FluxHeatmap extends React.Component<Props, State> {
 		const container = this.state.container;
```
Comment (eluchsinger) [inline src/containers/fluxHeatmap/FluxHeatmap.jsx]: These assignments could be summed up together like this: 
```jsx
const { container, maxLuxValue } = this.state;
const { transformation, heatmapMode } = this.props;
```

**Comment by eluchsinger** [comment]
Basic tests for some components are missing.

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx (hunk 1 above)
Comment (pacs01) [reply src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.jsx]: the parameter names are optional: https://flow.org/en/docs/types/functions/#toc-function-types

File: src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.test.jsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
+// @flow
+import * as React from "react";
+import { shallow } from "enzyme";
+import HeatmapAnalysisForm from "./HeatmapAnalysisForm";
+import type { AllInputTypes } from "../../../utils/InputHandler";
```
Comment (eluchsinger) [inline src/components/heatmap/heatmapAnalysisForm/HeatmapAnalysisForm.test.jsx]: (Wrong order)

**Pull request merged**